### PR TITLE
Add imports

### PR DESCRIPTION
**Pull request by joshbetz**
Adds two new commands:

```
$ vip import sql --help

  Usage: sql [options] <site> <file>

  Import SQL to a VIP Go site

  Options:

    -h, --help  output usage information
```

```
$ vip import files --help

  Usage: files [options] <site> <directory>

  Import files to a VIP Go site

  Options:

    -h, --help                output usage information
    -t, --types <types>       Types of files to import
    -p, --parallel <threads>  Number of parallel uploads
    -i, --intermediate        Upload intermediate images
    -f, --fast                Skip existing file check
```

### CODE REVIEW

**Comment by joshbetz** [comment]
Still need to

* [ ] Add progress bar for files uploader
* [ ] Remove superagent dependency from files uploader

File: bin/vip-import.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+								return upload( file, cb );
+							} else {
+								request
+									.get( url )
+									.set({ 'X-Client-Site-ID': site.client_site_id })
+									.set({ 'X-Access-Token': access_token })
+									.set({ 'X-Action': 'file_exists' })
+									.timeout( 2000 )
+									.end( ( err, res ) => {
+										if ( res && res.notFound ) {
```
Comment (nickdaugherty) [inline bin/vip-import.js]: Superagent actually throws an error 4xx/5xx responses, so the recommended way to handle that is

```
if (err && err.status === 404) {
  alert('oh no ' + res.body.message);
}
```

https://visionmedia.github.io/superagent/#error-handling

File: bin/vip-import.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+				.end( ( err, res ) => {
+					if ( err ) {
+						return console.error( err.response.error );
+					}
+
+					var db = res.body,
+						sql = fs.readFileSync( file ).toString()
+							.split( /;(\r\n|\r|\n)/ )
+							.map( s => s.trim() )
+							.filter( s => s.length > 0 );
```
Comment (nickdaugherty) [inline bin/vip-import.js]: This is pretty cool - though I'm worried about reading the entire SQL file into memory at once, in terms of memory usage. 

Node provides a streaming file api, as well as a 'readline' api that can be used to easily step through the file, line by line, which would be much more memory efficient given that we don't use the full file for anything here.

http://stackoverflow.com/a/32599033/623854

File: src/utils.js [hunk 3]
```diff
@@ -76,6 +77,33 @@ var utils = {
...
+
+			if ( ! s ) {
+				return console.error( "Couldn't find site:", site );
+			}
+
+			console.log( "Client Site:", s.client_site_id );
+			console.log( "Primary Domain:", s.domain_name );
+			console.log( "Environment:", s.environment_name );
+
+			promptly.confirm( "Are you sure?", ( err, t ) => {
```
Comment (nickdaugherty) [inline src/utils.js]: What does `t` mean?

File: src/utils.js [hunk 4]
```diff
@@ -76,6 +77,33 @@ var utils = {
...
+				if ( err ) {
+					return console.error( err );
+				}
+
+				if ( ! t ) {
+					return;
```
Comment (nickdaugherty) [inline src/utils.js]: Will this cause the program to exit? Might be worth a comment indicating such, and even some user output.

**Comment by nickdaugherty** [comment]
@joshbetz this is awesome - left a few comments here and there, and tweaked a variable name (unrelated to this changeset, but required for ES5 transpiling for testing) + whitespace.

**Comment by nickdaugherty** [comment]
The new changes look good - I'll go ahead and merge, and we can keep working on better large file support.

**Comment by joshbetz** [comment]
Noting some of the advantages of doing this in Javascript:

* We don't need the MySQL CLI client installed to run imports
* Progress bar
* Potentially lower memory footprint if we can stream the file?

**Pull request merged**